src/persist-client/src/internal/apply.rs

```diff
@@ -60,14 +60,14 @@ impl<K, V, T: Clone, D> Clone for Applier<K, V, T, D> {
             metrics: Arc::clone(&self.metrics),
             shard_metrics: Arc::clone(&self.shard_metrics),
             state_versions: Arc::clone(&self.state_versions),
-            state: self.state.clone(
-                self.state.applier_version.clone(),
-                self.state.hostname.clone(),
-            ),
+            state: Arc::clone(&self.state),
```

Comment:
I wonder if we could remove that awful `clone(version, hostname)` method from State now